src/main/java/jenkins/plugins/git/GitSCMFileSystem.java

```diff
@@ -283,15 +284,66 @@ public boolean supportsDescriptor(SCMSourceDescriptor descriptor) {
             return AbstractGitSCMSource.class.isAssignableFrom(descriptor.clazz);
         }
 
+        public static class HeadNameResult
+        {
+            private final String headName;
+            private final String prefix;
+
+            public HeadNameResult(String headName, String prefix) {
```

Comment:
Stylistically I would recommend
```suggestion
            private HeadNameResult(String headName, String prefix) {
```
and then moving `calculateHeadName` inside this nested class, and renaming it accordingly (to simply `calculate`, or `create`, or `make`, or `of`, etc.).